### ISSUE TITLE
githubfs

### ISSUE THREAD

**Issue by mariusvniekerk** [regro/cf-scripts#1060]
As we start using github more like a fillesystem in many of the places in this repo, it would be nice to have a fsspec based GithubFs (likely backed by aiohttp)

This would allow us to do something like

```
async with fs.open(f'github://{org}/{repo}/{path}') as fo:
    content = await fo.read()
```

This would allow us to relatively easily build index files for all the variants we generate and be able to retrieve that file statically. (say) `.ci_support/variants.index.json` and then loop through the files and retrieve them using our filesystem.

Odds are we want to make this read-only

**Comment by beckermr** [comment]
Oh interesting. I was going to download the zipped copy of the repo that GitHub provides for master to do this.

**Comment by mariusvniekerk** [comment]
Yeah, the zipped copy download will probably scale pretty badly and get us rate limited fast.

**Comment by beckermr** [comment]
how is that download different from the raw download we use now?

**Comment by mariusvniekerk** [comment]
its not, the idea is just to make a nice wrapper around the raw content github api, we generally don't need "all the files"

**Comment by beckermr** [comment]
ah I thought you meant it would be more rate-limited than we are now
